packages/dataviews/src/types.ts

```diff
@@ -292,6 +298,11 @@ interface ViewBase {
 	 * The fields to render
 	 */
 	fields?: string[];
+
+	/**
+	 * The density of the view.
+	 */
+	density?: Density;
```

Comment:
API-wise, I think this is the right choice.

A DataViews component prop seems a bit detached from what we already have (view config, default layouts).

`view` vs `view.layout`: density is a tool that DataViews shows to users in the view config. The `view.layout` is about things that are layout-specific and that the rest of the DataViews component doesn't know anything about (e.g.: the logic to handle the media field of grid is contained within the layout itself, there's no other controls in DataViews affected by that).